src/core.js

```diff
@@ -23,32 +22,65 @@ async function loadPage (page, url, timeout, pageLoadSkipTimeout) {
   // set a higher number than the timeout option, in order to make
   // puppeteer’s timeout _never_ happen
   let waitingForPageLoad = true
-  const loadPagePromise = page.goto(url, { timeout: timeout + 1000 })
+  let pageLoadSkipTimeoutPromise = null
+  const loadPagePromise = new Promise((resolve, reject) => {
+    page
+      .goto(url, {
+        timeout: timeout + 1000
+      })
+      .then(response => {
+        // Reject exits the Promise.all call immediately and ensures that the pageload is always higher prio than timout
+        return resolve('loadPageResponse')
+      })
+      .catch(err => {
+        // Reject when error because we don't want an errored page
+        console.error(err)
+        return reject(err)
+      })
```

Comment:
I'm not sure this `then...catch` code does anything, but tell me if you think I'm wrong! The error would have rejected the promise by itself, so catching and manually rejecting it should be redundant. Similarly, you are not making use of the explicit `loadPageResponse` value that you added, hence that `thenable` seem unnecessary.

Reply:
You are right. This was a leftover of my previous version. Will change it later on